src/sql/parts/disasterRecovery/restore/common/restoreServiceImpl.ts

```diff
@@ -153,9 +155,15 @@ export class RestoreDialogController implements IRestoreDialogController {
 			restoreOption.taskExecutionMode = TaskExecutionMode.executeAndScript;
 		}
 
-		this._restoreService.restore(this._ownerUri, restoreOption);
-		let restoreDialog = this._restoreDialogs[this._currentProvider];
-		restoreDialog.close();
+		this._restoreService.restore(this._ownerUri, restoreOption).then(() => {
```

Comment:
restore creates a tasks so when the promise response is back, it doesn't mean the database is restored. it mean the task is created. you need to do this in the task manager looking for restore tasks and when they are complete, refresh  the tree node

Reply:
I've rewritten this part to use the taskService now.